FrontBundle/Routing/Database/OpenOrchestraDatabaseRouter.php

```diff
@@ -0,0 +1,178 @@
+<?php
+
+namespace OpenOrchestra\FrontBundle\Routing\Database;
+
+use OpenOrchestra\FrontBundle\Manager\NodeManager;
+use OpenOrchestra\FrontBundle\Routing\Database\Transformer\RouteDocumentCollectionToRouteCollectionTransformer;
+use OpenOrchestra\FrontBundle\Routing\Database\Transformer\RouteDocumentToValueObjectTransformer;
+use OpenOrchestra\ModelInterface\Repository\RouteDocumentRepositoryInterface;
+use Symfony\Component\HttpFoundation\RequestStack;
+use Symfony\Component\Routing\Exception\InvalidParameterException;
+use Symfony\Component\Routing\Exception\MethodNotAllowedException;
+use Symfony\Component\Routing\Exception\MissingMandatoryParametersException;
+use Symfony\Component\Routing\Exception\ResourceNotFoundException;
+use Symfony\Component\Routing\Exception\RouteNotFoundException;
+use Symfony\Component\Routing\Generator\UrlGeneratorInterface;
+use Symfony\Component\Routing\Matcher\UrlMatcherInterface;
+use Symfony\Component\Routing\RequestContext;
+use Symfony\Component\Routing\RouteCollection;
+use Symfony\Component\Routing\RouterInterface;
+
+/**
+ * Class OpenOrchestraDatabaseRouter
+ */
+class OpenOrchestraDatabaseRouter implements RouterInterface
+{
+    /**
+     * @var RequestContext
+     */
+    protected $context;
+
+    protected $routeDocumentCollectionToRouteCollectionTransformer;
+    protected $routeDocumentToValueObjectTransformer;
+    protected $routeDocumentRepository;
+    protected $options = array();
+    protected $routeCollection;
+    protected $requestStack;
+    protected $nodeManager;
+    protected $generator;
+    protected $matcher;
+
+    /**
+     * @param RouteDocumentRepositoryInterface                    $routeDocumentRepository
+     * @param RouteDocumentToValueObjectTransformer               $routeDocumentToValueObjectTransformer
+     * @param RouteDocumentCollectionToRouteCollectionTransformer $routeDocumentCollectionToRouteCollectionTransformer
+     * @param RequestStack                                        $requestStack
+     * @param NodeManager                                         $nodeManager
+     * @param array                                               $options
+     */
+    public function __construct(
+        RouteDocumentRepositoryInterface $routeDocumentRepository,
+        RouteDocumentToValueObjectTransformer $routeDocumentToValueObjectTransformer,
+        RouteDocumentCollectionToRouteCollectionTransformer $routeDocumentCollectionToRouteCollectionTransformer,
+        RequestStack $requestStack,
+        NodeManager $nodeManager,
+        array $options = array()
+    )
+    {
+        $this->routeDocumentCollectionToRouteCollectionTransformer = $routeDocumentCollectionToRouteCollectionTransformer;
```

Comment:
I think the var name 'routeDocumentCollectionToRouteCollectionTransformer' is too long